src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -6112,24 +6121,44 @@ protected override bool IsItemContainer(string path)
         /// <param name="force">If true, force move the directory, overwriting anything at the destination.</param>
         private void MoveDirectoryInfoUnchecked(DirectoryInfo directory, string destinationPath, bool force)
         {
-            try
+            if (!IsSameWindowsVolume(directory.FullName, destinationPath))
             {
-                if (InternalTestHooks.ThrowExdevErrorOnMoveDirectory)
-                {
-                    throw new IOException("Invalid cross-device link");
-                }
-
-                directory.MoveTo(destinationPath);
+                CopyAndDelete(directory, destinationPath, force);
```

Comment:
It makes no sense to do the check here since it doesn't cover all scenarios - paths with different root directory can point to the same volume and vise verse.
We should trust OS behavior and do directory.MoveTo() first. Then handle exception in right way. See my comment https://github.com/PowerShell/PowerShell/issues/16000#issuecomment-906926525

Reply:
.NET runtime has the same check, the alternative is to check another HResult on Windows to catch that case